package/MDAnalysis/core/groups.py

```diff
@@ -822,10 +822,10 @@ def _asunique(self, group, sorted=False, set_mask=False):
 
         indices = unique_int_1d_unsorted(self.ix)
         if set_mask:
-            mask = np.zeros_like(self.ix)
-            for i, x in enumerate(indices):
-                values = np.where(self.ix == x)[0]
-                mask[values] = i
+            # self.ix must be contiguous for cython implementation.
+            # crashes on slices of atomgroups otherwise.
+            mask = inverse_unique_contiguous_1d_array(
+                       np.ascontiguousarray(self.ix), indices)
```

Comment:
I'd just push the ascontiguous call into the function rather than leave the sharp edge dangling